spec.html

```diff
@@ -6824,6 +6814,27 @@ <h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-getiterator" type="abstract operation">
+      <h1>
+        GetIterator (
+          _obj_: an ECMAScript language value,
+          _kind_: ~sync~ or ~async~,
+        ): either a normal completion containing an Iterator Record or a throw completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. If _kind_ is ~async~, then
+          1. Let _method_ be ? GetMethod(_obj_, @@asyncIterator).
+          1. If _method_ is *undefined*, then
+            1. Let _syncMethod_ be ? GetMethod(_obj_, @@iterator).
+            1. Let _syncIteratorRecord_ be ? GetIteratorFromMethod(_obj_, _syncMethod_).
+            1. Return CreateAsyncFromSyncIterator(_syncIteratorRecord_).
+        1. Otherwise, let _method_ be ? GetMethod(_obj_, @@iterator).
```

Comment:
I'm pretty sure that, for both occurrences of `GetMethod(_obj_, @@iterator)`, the result might be `*undefined*`, which isn't being handled, it's just being passed to the second parameter of `GetIteratorFromMethod`, which is expecting a function object.

(This is a pre-existing problem, but you might want to fix it in this PR.)